lib/src/io.dart

```diff
@@ -963,10 +962,10 @@ ByteStream createTarGz(List<String> contents, {String baseDir}) {
   return new ByteStream(StreamCompleter.fromFuture(new Future.sync(() async {
     var buffer = new StringBuffer();
     buffer.write('Creating .tar.gz stream containing:\n');
-    contents.forEach((file) => buffer.write('$file\n'));
+    buffer.writeAll(contents, '\n');
```

Comment:
technically this is not emitting the last newline that would have been emitted before

Reply:
Switched to `contents.forEach(buffer.writeln);`